### PR TITLE
Search client by name

### PR DESCRIPTION
**Pull request by guy-ro**
See https://www.notion.so/uwblueprintexecs/Server-Find-client-when-creating-updating-request-d460bc2d300046a79a53fc74ba9e2ee5

We needed a way for the user of the API to query clients by their fullName. To generalize this:

- new endpoint called filterClients that takes as input a "filter" object that has all the parameters of a client
- this endpoint returns an array of Clients, filtered by whichever properties of the filter object are defined

This generalization implements filtering on any property of a Client.

I left commented out a second working approach. Not sure which approach is best.

E.g:
image

### CODE REVIEW

File: server/graphql/resolvers.ts [hunk 1]
```diff
@@ -41,6 +41,20 @@ const nextRequestRequestGroupHelper = (requestTypeIds, dataSources): RequestInte
 const resolvers = {
   Query: {
     client: (_, { id }, { dataSources }): ClientInterface => dataSources.clients.getById(Types.ObjectId(id)),
+    filterClients: (_, { filter }, { dataSources }): Array<ClientInterface> => {
+      let filteredClients = dataSources.clients.getAll()
+
+      for (let property in filter) {
+        filteredClients = filteredClients.filter((client) => (client[property] ? client[property] === filter[property] : true))
+      }
```
Comment (meganniu) [inline server/graphql/resolvers.ts:49]: I like the uncommented approach better :)) more maintainable

**Review by meganniu — APPROVED** [review]
LGTM, thanks!!!!!

File: server/graphql/schema.ts [hunk 2]
```diff
@@ -105,6 +106,7 @@ const typeDefs = gql`
         id: ID
         clientId: String
         fullName: String
+        deleted: Boolean
```
Comment (jlight99) [inline server/graphql/schema.ts:109]: why are we adding deleted? is this related to the filtering logic?

**Review by jlight99 — APPROVED** [review]
lgtm! had one question that i left in a comment

File: server/graphql/schema.ts (hunk 2 above)
Comment (guy-ro) [reply server/graphql/schema.ts:109]: Will address this in a new PR (including changes discussed on Slack)
